compiler/catala_utils/pos.mli

```diff
@@ -18,11 +18,14 @@
 
 type t
 (** A position in the source code is a file, as well as begin and end location
-    of the form col:line *)
+    of the form col:line, and optionally, attributes *)
 
 val compare : t -> t -> int
 val equal : t -> t -> bool
 
+type attr = ..
+type attr += Law_pos of string list
```

Comment:
I'm not against it but why cramming the attributes in the positions instead of the `mark` type of the AST?

Reply:
Marks are only on expressions. Positions are everywhere. And attributes are actually mostly expected to be useful on types, top-level declarations, structure fields, etc.
Now, indeed, we could rename `Pos` to something else (and maybe use raw positions inside of attributes instead of positions that could at the moment recursively embed attributes, although we make no use of that). But that would make the changes bigger and the actual change more difficult to review...

Reply:
To make myself clearer, I was envisionning a modification to 

https://github.com/CatalaLang/catala/blob/b15b5d11c4082f0626b48e26bd02a252acf6f136/compiler/shared_ast/definitions.ml#L431-L446

to add the attributes there instead of directly into `Pos.t`.